src/Pos/Wallet/Web/ClientTypes/Types.hs

```diff
@@ -0,0 +1,572 @@
+-- you will effectively get /change/ addresses.
+isTxLocalAddress
+    :: [CWAddressMeta]      -- ^ All addresses in wallet
+    -> NonEmpty (CId Addr)  -- ^ Input addresses of transaction
+    -> (CId Addr -> Bool)
+isTxLocalAddress wAddrMetas inputs = do
+    let mLocalAddrs = getTxSourceAccountAddresses wAddrMetas inputs
+    case mLocalAddrs of
+       Just changeAddrs -> do
+           -- if given wallet is source of tx, /local addresses/
+           -- can be fetched according to definition
+           let changeAddrsSet = S.fromList changeAddrs
+           flip S.member changeAddrsSet
+       Nothing -> do
+           -- if given wallet is *not* source of tx, then it's incoming
+           -- transaction, and only addresses of given wallet are *not*
+           -- local addresses
+           -- [CSM-309] This may work until transaction have multiple
+           -- destination addresses
+           let nonLocalAddrsSet = S.fromList $ cwamId <$> wAddrMetas
+           not . flip S.member nonLocalAddrsSet
+
+mkCTxs
+    :: ChainDifficulty    -- ^ Current chain difficulty (to get confirmations)
+    -> TxHistoryEntry     -- ^ Tx history entry
+    -> CTxMeta            -- ^ Transaction metadata
+    -> [CWAddressMeta]    -- ^ Addresses of wallet
+    -> Either Text CTxs
+mkCTxs diff THEntry {..} meta wAddrMetas = do
+    let isOurTxOutput = flip S.member wAddrsSet . addressToCId . txOutAddress
+
+        ownInputs = filter isOurTxOutput inputs
+        ownOutputs = filter isOurTxOutput outputs
+
+    when (null ownInputs && null ownOutputs) $
+        throwError "Transaction is irrelevant to given wallet!"
+
+    let sumMoney = sumCoins . map txOutValue
+        outgoingMoney = sumMoney ownInputs
+        incomingMoney = sumMoney ownOutputs
+        isOutgoing = outgoingMoney >= incomingMoney
+        isIncoming = incomingMoney >= outgoingMoney
+
+        ctAmount = mkCCoin . unsafeIntegerToCoin $
+            if | isOutgoing && isIncoming -> outgoingMoney
+               | isOutgoing -> outgoingMoney - incomingMoney
+               | isIncoming -> incomingMoney - outgoingMoney
+
+        mkCTx ctIsOutgoing cond = guard cond $> CTx {..}
+        ctsOutgoing = mkCTx True isOutgoing
+        ctsIncoming = mkCTx False isIncoming
+    return CTxs {..}
+  where
+    ctId = txIdToCTxId _thTxId
+    inputs = _thInputs
+    outputs = toList $ _txOutputs _thTx
+    ctInputAddrs = map addressToCId _thInputAddrs
+    ctOutputAddrs = map addressToCId _thOutputAddrs
+    ctConfirmations = maybe 0 fromIntegral $ (diff -) <$> _thDifficulty
+    ctMeta = meta
+    wAddrsSet = S.fromList $ map cwamId wAddrMetas
+
+newtype CPassPhrase = CPassPhrase Text
+    deriving (Eq, Generic)
+
+instance Show CPassPhrase where
+    show _ = "<pass phrase>"
+
+type instance OriginType CPassPhrase = PassPhrase
+
+-- These two instances nearly duplicate `instance Bi PassPhrase`
+-- in `Pos.Binary.Crypto`.
+instance FromCType CPassPhrase where
```

Comment:
Why not move these instances (and all others in this file) to `Pos.Wallet.Web.ClientTypes.Instances`?